src/runtime/methodbinder.cs

```diff
@@ -301,192 +299,242 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                     isGeneric = true;
                 }
                 ParameterInfo[] pi = mi.GetParameters();
-                var clrnargs = pi.Length;
-                var match = false;
-                var arrayStart = -1;
+                ArrayList defaultArgList;
+                bool paramsArray;
+
+                if (!MatchesArgumentCount(pynargs, pi, out paramsArray, out defaultArgList)) {
+                    continue;
+                }
                 var outs = 0;
+                var margs = TryConvertArguments(pi, paramsArray, args, pynargs, defaultArgList,
+                    needsResolution: _methods.Length > 1,
+                    outs: out outs);
 
-                if (pynargs == clrnargs)
+                if (margs == null)
                 {
-                    match = true;
+                    continue;
                 }
-                else if (pynargs < clrnargs)
+
+                object target = null;
+                if (!mi.IsStatic && inst != IntPtr.Zero)
                 {
-                    match = true;
-                    defaultArgList = new ArrayList();
-                    for (var v = pynargs; v < clrnargs; v++)
+                    //CLRObject co = (CLRObject)ManagedType.GetManagedObject(inst);
+                    // InvalidCastException: Unable to cast object of type
+                    // 'Python.Runtime.ClassObject' to type 'Python.Runtime.CLRObject'
```

Comment:
This comment confuses me more than it helps ;). It's clear after reading the next one, but here not so much.